packages/openapi3-parser/lib/parser/oas/parseOpenAPIObject.js

```diff
@@ -49,14 +51,18 @@ function parseOASObject(context, object) {
     // FIXME Support exposing extensions into parse result
     [isExtension, () => new namespace.elements.ParseResult()],
 
+    [
+      R.both(hasKey('webhooks'), isOpenAPI31OrHigher),
+      createUnsupportedMemberWarning(namespace, name),
+    ],
     [isUnsupportedKey, createUnsupportedMemberWarning(namespace, name)],
 
     // Return a warning for additional properties
     [R.T, createInvalidMemberWarning(namespace, name)],
   ]);
 
   const parseOASObject = pipeParseResult(namespace,
-    parseObject(context, name, parseMember, requiredKeys, ['components']),
+    parseObject(context, name, parseMember, requiredKeys, ['openapi', 'components']),
```

Comment:
To aid review, the array of keys passed to `parseObject` here signify an order of keys that should be parsed upfront (in order). This means that `openapi` is the first key to be parsed as other code will depend on it being parsed. Thus for documents such as the following:

```
webhooks: {}
openapi: 3.1.0
```

The parse order will actually be `openapi` then `webhooks`.